### PR TITLE
SIG-Network: Add Subproject Leads responsibilities

### PR DESCRIPTION
**Pull request by tssurya**
**Which issue(s) this PR fixes**:

1. Add more structure to what it means to being a sub-project lead.
2. Add sub project leads and owner repos to existing areas

### CODE REVIEW

File: sig-network/README.md [hunk 1]
```diff
@@ -134,6 +134,25 @@ The following [subprojects][subproject-definition] are owned by sig-network:
...
 <!-- BEGIN CUSTOM CONTENT -->
+
+### Responsibilities of Subproject Leads
+
+Just like the [SIG Network Chair's](https://github.com/kubernetes/community/blob/master/sig-network/role-sig-chair.md#whats-expected) expectation is to "run the show" of the whole SIG, the Subproject Lead's expectation is to "run the show" for that subproject. This includes:
+
+- **Transparent Project Planning and Maintainence:**
+  - Leads must provide transparent view into their historical and future project plans (eg; using GitHub project boards)
+  - Leads must communicate the project plans to the wider community by linking it to the project's README pages.
+  - Project must be kept in good health by the lead through regular issue triaging and PR reviews (delegating this to other members in the community is good!)
```
Comment (aojea) [inline sig-network/README.md:145]: Check testgrids and jobs failures, CI Health

File: sig-network/README.md [hunk 2]
```diff
@@ -134,6 +134,25 @@ The following [subprojects][subproject-definition] are owned by sig-network:
...
+
+- **Regular Project Updates**
+  - Leads must report on project's status, significant releases/events, interesting developments to the wider SIG-Network community (eg; in the SIG-Network meetings or sig-network slack channel or mailing-list).
```
Comment (aojea) [inline sig-network/README.md:148]: Projects that includes the official CRDs using k8s.io  must go throught API review process and is described here https://github.com/kubernetes/community/blob/master/sig-architecture/api-review-process.md

Leads are expected to make sure this well defined processes across the project are followed

File: sig-network/README.md (hunk 1 above)
Comment (tssurya) [reply sig-network/README.md]: also good point, but I see all these points got overwritten for the final version; I can add this back in @shaneutt are you ok with adding this and the api-review bits that antonio suggested into the list?

File: sig-network/README.md [hunk 3]
```diff
@@ -134,6 +149,29 @@ The following [subprojects][subproject-definition] are owned by sig-network:
 [subproject-definition]: https://github.com/kubernetes/community/blob/master/governance.md#subprojects
 [working-group-definition]: https://github.com/kubernetes/community/blob/master/governance.md#working-groups
 <!-- BEGIN CUSTOM CONTENT -->
+
+### Subproject Leads
+
+SIG network provides some additional responsibilities for subproject leads beyond was in covered in the [standard subproject definition][subprojects].
+Most of these additional responsibilities relate to communication about their projects within SIG Network, and to the greater community:
+
+- **Transparent Project Planning, Maintainence and Communication:**
```
Comment (tssurya) [inline sig-network/README.md:158]:
```suggestion
- **Transparent Project Planning, Maintenence and Communication:**
```

File: sig-network/README.md (hunk 1 above)
Comment (shaneutt) [reply sig-network/README.md]: Sure np

File: sig-network/README.md (hunk 2 above)
Comment (shaneutt) [reply sig-network/README.md]: Yep, let's add that.

File: sig-network/README.md [hunk 4]
```diff
@@ -90,6 +90,10 @@ The following [subprojects][subproject-definition] are owned by sig-network:
 - **Contact:**
   - Slack: [#external-dns](https://kubernetes.slack.com/messages/external-dns)
 ### gateway-api
+- **Leads:**
+  - Rob Scott (**[@robscott](https://github.com/robscott)**), Google
+  - Shane Utt (**[@shaneutt](https://github.com/shaneutt)**), Kong
+  - Nick Young (**[@youngnick](https://github.com/youngnick)**), Isovalent
```
Comment (shaneutt) [inline sig-network/README.md:96]: Since we started working on this doc @tssurya, @mlavacca has joined as a Gateway API lead and I've switched orgs to Red Hat:

```suggestion
- **Leads:**
  - Mattia Lavacca (**[@mlavacca](https://github.com/mlavacca)**), Kong
  - Rob Scott (**[@robscott](https://github.com/robscott)**), Google
  - Shane Utt (**[@shaneutt](https://github.com/shaneutt)**), Red Hat
  - Nick Young (**[@youngnick](https://github.com/youngnick)**), Isovalent
```

File: sigs.yaml [hunk 5]
```diff
@@ -2264,6 +2264,16 @@ sigs:
...
+    leads:
+    - github: robscott
+      name: Rob Scott
+      company: Google
+    - github: shaneutt
+      name: Shane Utt
+      company: Kong
+    - github: youngnick
+      name: Nick Young
+      company: Isovalent
```
Comment (shaneutt) [inline sigs.yaml:2276]: Similar to above:

```suggestion
    leads:
    - github: mlavacca
      name: Mattia Lavacca
      company: Kong
    - github: robscott
      name: Rob Scott
      company: Google
    - github: shaneutt
      name: Shane Utt
      company: Red Hat
    - github: youngnick
      name: Nick Young
      company: Isovalent
```

File: sig-network/README.md (hunk 4 above)
Comment (tssurya) [reply sig-network/README.md:96]: ack resolving this

File: sig-network/README.md [hunk 6]
```diff
@@ -90,6 +90,11 @@ The following [subprojects][subproject-definition] are owned by sig-network:
 - **Contact:**
   - Slack: [#external-dns](https://kubernetes.slack.com/messages/external-dns)
 ### gateway-api
+- **Leads:**
+  - Mattia Lavacca (**[@mlavacca](https://github.com/mlavacca)**), Kong
+  - Rob Scott (**[@robscott](https://github.com/robscott)**), Google
+  - Shane Utt (**[@shaneutt](https://github.com/shaneutt)**), Kong
```
Comment (shaneutt) [inline sig-network/README.md:96]:
```suggestion
  - Shane Utt (**[@shaneutt](https://github.com/shaneutt)**), Red Hat
```

File: sigs.yaml [hunk 7]
```diff
@@ -2260,6 +2260,19 @@ sigs:
...
+    leads:
+    - github: mlavacca
+      name: Mattia Lavacca
+      company: Kong
+    - github: robscott
+      name: Rob Scott
+      company: Google
+    - github: shaneutt
+      name: Shane Utt
+      company: Kong
```
Comment (shaneutt) [inline sigs.yaml:2272]:
```suggestion
      company: Red Hat
```

File: sig-network/README.md (hunk 6 above)
Comment (tssurya) [reply sig-network/README.md:96]: shame on me

File: sig-network/README.md [hunk 8]
```diff
@@ -135,6 +148,31 @@ The following [subprojects][subproject-definition] are owned by sig-network:
...
 <!-- BEGIN CUSTOM CONTENT -->
+
+### Subproject Leads
+
+SIG network provides some additional responsibilities for subproject leads beyond was in covered in the [standard subproject definition][subprojects].
+Most of these additional responsibilities relate to communication about their projects within SIG Network, and to the greater community:
+
+- **Transparent Project Planning, Maintenance and Communication:**
+  - Subproject Leads **MUST** provide transparent view into their historical and future project plans (eg; using GitHub project boards, KEPs, or custom enhancement proposals (see [GEPs]/[NPEPs])).
+  - Subproject leads **MUST** create, maintain and be present in a public [Kubernetes Slack] channel with the naming `#sig-network-<subproject>`.
```
Comment (shaneutt) [inline sig-network/README.md:159]:
```suggestion
  - Subproject leads **MUST** create, maintain and be present in a public [Kubernetes Slack] channel with the easy-to-find naming `#sig-network-<subproject>` for their sub-project.
```

File: sig-network/README.md [hunk 9]
```diff
@@ -135,6 +148,31 @@ The following [subprojects][subproject-definition] are owned by sig-network:
...
+  - Subproject leads **SHOULD** create and maintain a regular public [Zoom] sync on the [SIG Network Calendar].
+  - Subproject leads **MUST** keep the project in good health through regular issue triaging, PR reviews, CI health monitoring and checking testgrids (delegating this to other members in the community is good!)
```
Comment (shaneutt) [inline sig-network/README.md:161]: While it's absolutely true that we want subprojects to do this, this one is difficult.

With the above MUST items (make your process clear, and create a channel) these are easy things to identify when they aren't happening and just point to this doc. "good health" for a project however can be really subjective :thinking: I think I'm in favor of dropping this one for now, if we feel in the future there's some need related to this I think it should be its own effort.

```suggestion
```

File: sig-network/README.md [hunk 10]
```diff
@@ -135,6 +148,31 @@ The following [subprojects][subproject-definition] are owned by sig-network:
...
+  - Projects that include the official CRDs using `k8s.io` must go through API review process as is described here https://github.com/kubernetes/community/blob/master/sig-architecture/api-review-process.md and Subproject leads are expected to make sure this well defined processes across Kubernetes is followed.
```
Comment (shaneutt) [inline sig-network/README.md:162]: Unfortunately "official CRDs" is only a colloquialism at the moment, I recommend this language instead:

```suggestion
  - Projects that distribute CRDs **MUST** go through the [API review process](https://github.com/kubernetes/community/blob/master/sig-architecture/api-review-process.md) for major releases.
```

File: sigs.yaml [hunk 11]
```diff
@@ -2259,16 +2272,32 @@ sigs:
...
     contact:
       slack: sig-network-policy-api
     owners:
     - https://raw.githubusercontent.com/kubernetes-sigs/kube-network-policies/master/OWNERS
     - https://raw.githubusercontent.com/kubernetes-sigs/network-policy-api/master/OWNERS
     - https://raw.githubusercontent.com/kubernetes/api/master/networking/OWNERS
+    leads:
+    - github: astoycos
+      name: Andrew Stoycos
+      company: Red Hat
```
Comment (danwinship) [inline sigs.yaml:2294]: nope :slightly_frowning_face:

File: sig-network/README.md (hunk 10 above)
Comment (shaneutt) [reply sig-network/README.md:162]: Indeed, it was meant as more of an extra reminder but if you feel its too redundant I'm fine with dropping it.

File: sig-network/README.md [hunk 12]
```diff
@@ -135,6 +148,31 @@ The following [subprojects][subproject-definition] are owned by sig-network:
...
 [working-group-definition]: https://github.com/kubernetes/community/blob/master/governance.md#working-groups
 <!-- BEGIN CUSTOM CONTENT -->
+
+### Subproject Leads
+
+SIG network provides some additional responsibilities for subproject leads beyond was in covered in the [standard subproject definition][subprojects].
+Most of these additional responsibilities relate to communication about their projects within SIG Network, and to the greater community:
+
+- **Transparent Project Planning, Maintenance and Communication:**
+  - Subproject Leads **MUST** provide transparent view into their historical and future project plans (eg; using GitHub project boards, KEPs, or custom enhancement proposals (see [GEPs]/[NPEPs])).
```
[the remark this answers is not in the document]
Comment (shaneutt) [reply sig-network/README.md:158]: Sorry, habit. Inspired by RFCs and a common [thing we've done in Gateway API](https://github.com/kubernetes-sigs/gateway-api/blob/ce6d2c6366847386747d7aa6ebb8ff95693dd536/apis/v1/gateway_types.go#L70), similarly inspired. It's a good callout that if someone comes in here and doesn't know that, it might seem a bit nasty. I'm fine with just changing all these to a plain must.

File: sigs.yaml (hunk 11 above)
Comment (shaneutt) [reply sigs.yaml:2294]: True, @astoycos what shall we put in here?

File: sig-network/README.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (danwinship) [reply sig-network/README.md:158]: In RFCs it's always applied to _implementations_ though. It feels weird to apply it to people.

[the remark this answers is not in the document]
Comment (shaneutt) [reply sig-network/README.md:158]: Agreed

[the remark this answers is not in the document]
Comment (tssurya) [reply sig-network/README.md:158]: ack changing it to must

File: sig-network/README.md (hunk 9 above)
Comment (tssurya) [reply sig-network/README.md:161]: hmm but what's the point of being a lead if your project members are not having a good experience... so what I meant in this point with issue triage and PR reviews is that I am not necessarily saying the lead has to be the one doing it; but the lead needs to make sure "its happening somehow ;; either the members or rotating with other leads etc..."

the CI health monitoring and testgrid was @aojea 's suggestion I'll let him chime in

File: sig-network/README.md (hunk 10 above)
Comment (tssurya) [reply sig-network/README.md:162]: this was suggested by @aojea as well from previous review iterations.. will wait to see what he thinks..

Comment (aojea) [reply sig-network/README.md:162]: I saw some confusion around this several times so I thought it was good to reinforce the message

Comment (shaneutt) [reply sig-network/README.md:162]: Dan as long as you're not strongly opposed, then I propose we leave this in as an extra reminder?

File: sig-network/README.md (hunk 9 above)
Comment (shaneutt) [reply sig-network/README.md]: In retrospect, I rescind my thought about dropping this. Let's just use the softer must language and consider this resolved.

**Comment by jberkus** [comment]
@shaneutt @aojea hey, are y'all ever merging this?

**Comment by shaneutt** [comment]
> @shaneutt @aojea hey, are y'all ever merging this?

(I was one of the authors, so I'm probably not going to merge it lol)

**Comment by shaneutt** [comment]
ping!

File: sig-network/README.md (hunk 10 above)
Comment (aojea) [reply sig-network/README.md:162]: I was chatting with @robscott about this recently, I think that we need to reword this as something 

> Subprojects should follow project guidelines and involve the SIG in user-facing changes, such as new APIs, which require API review.

Comment (aojea) [reply sig-network/README.md:162]: @thockin please chime in and see if we can get this merged

Comment (thockin) [reply sig-network/README.md:162]: Anything with a *k8s.io prefix MUST go thru API review.  Other sub-projects SHOULD.  If I recall the rules correctly.

File: sigs.yaml (hunk 11 above)
Comment (thockin) [reply sigs.yaml:2298]: News to me.  :(

File: sig-network/README.md (hunk 10 above)
Comment (shaneutt) [reply sig-network/README.md:162]: That being the case, my preference would be that we add an explicit MUST for any official SIG Network sub-project that distributes CRD APIs. This helps us to (at bare minimum) stay in touch with what the project is up to (albeit the other stuff here about checking in regularly should help with that as well). One of the high level goals of adding these additional responsibilities is to promote more communication.

Comment (thockin) [reply sig-network/README.md:162]: I am fine with that.

**Comment by tssurya** [comment]
I will rebase this and fix the suggestions mentioned in the last series: https://github.com/kubernetes/community/pull/7967#discussion_r1791652286
we are getting close...

File: sigs.yaml (hunk 11 above)
Comment (tssurya) [reply sigs.yaml:2298]: @astoycos shall I put Citadel?

**Comment by adrianmoisey** [comment]
From a very-new-to-the-community perspective, the responsibilities documented here seem reasonable and communicated clearly.

File: sig-network/README.md (hunk 10 above)
Comment (tssurya) [reply sig-network/README.md:162]: tried to update it based on the discussion here, resolving this let's see if the next push makes it better.

**Comment by tssurya** [comment]
rebased! I think this should be ready now...

File: sigs.yaml (hunk 11 above)
Comment (tssurya) [reply sigs.yaml:2308]: we can revisit this later when @astoycos gets back to us

Comment (aojea) [reply sigs.yaml:2308]: I think is better to wait, it is ok if he can dedicate only X hours per week/month, but if he is not be able to participate at all it is better to reconsider his commitment, adding a lead that is not going to be present because of reasons is not great

**Comment by aojea** [comment]
/lgtm

modulo the comment on https://github.com/kubernetes/community/pull/7967#discussion_r1796692807

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

File: sigs.yaml (hunk 11 above)
Comment (astoycos) [reply sigs.yaml:2308]: Supper sorry For the late response here things are finally starting to settle in for me at the new gig and I'm just starting to catch back up.

So for now if we could just leave my "company" empty that would the best.  I think my available time is going to definitely be much smaller now :/  shooting for around 2 hours a week (usually on the weekends)

Ultimately I would like @tssurya to step in and take over my positions, she already has keep everything rolling over the past month while I've been MIA, and has been super actively involved for over a year now.

**Comment by astoycos** [comment]
Again I just want to apologize to everyone here for my recent lack of involvement, excited to see this role become cemented,  LGTM 👍

File: sigs.yaml (hunk 11 above)
Comment (aojea) [reply sigs.yaml:2308]: no need to remove anyone that still have the willingness to participate and still can dedicate some time ...

Comment (tssurya) [reply sigs.yaml:2308]: ack I will update the list to include myself if everyone else is OK with that and continue to keep @astoycos with empty company for now. Please let me know if anyone has issues with me doing that...

@astoycos +1 to what @aojea said, any small amount of time you put in is valuable!!

Comment (aojea) [reply sigs.yaml:2308]: I think instead of leaving it empty, we can use "Independent" , that is used in other places for people that does not identify with one particular company

Comment (tssurya) [reply sigs.yaml:2308]: ah actually I will update the list to have Andrew in it for now with empty company, given I am not in OWNERS file, its probably not appropriate to add myself till we get that fixed first.

**Comment by tssurya** [comment]
New changes are:

- removed andrew's company
- rebased on latest master
